### DISCUSSION TITLE
How do you use your BrowserFS local filesystem module?

### DISCUSSION THREAD

**Discussion by hello-smile6** [DustinBrett/daedalOS#115] [Q&A] [answered]
How do you use your BrowserFS local filesystem module?

**Comment by DustinBrett** [comment, marked as answer]
Hey, you can check out the `useFileSystem` hook at https://github.com/DustinBrett/daedalOS/blob/main/contexts/fileSystem/useFileSystemContextState.ts#L341

If you mean how did I implement it, that is here https://github.com/DustinBrett/daedalOS/blob/main/contexts/fileSystem/useAsyncFs.ts

**Reply by hello-smile6** [reply]
Let me try using the directory handle in an object...

**Reply by DustinBrett** [reply]
BrowserFS mocks the node fs for the most part. If you want you can get direct access to `fs`.

**Reply by hello-smile6** [reply]
Should I access it as `BrowserFS.FileSystem.FileSystemAccess.Create()`?

**Reply by DustinBrett** [reply]
It depends what you want to do, but for the most part you gotta play around with it.

https://github.com/jvilk/BrowserFS

**Reply by hello-smile6** [reply]
Okay, I'm using https://github.com/jvilk/BrowserFS/pull/321

**Reply by DustinBrett** [reply]
Ok then the API for that is here https://developer.mozilla.org/en-US/docs/Web/API/File_System_Access_API

It's possible there still may be issues, such as https://github.com/DustinBrett/daedalOS/issues/75

**Reply by hello-smile6** [reply]
Okay, how do I make BrowserFS use it as a backend?

**Reply by DustinBrett** [reply]
I'm not sure what it is you want to do but it sounds like you want to code something custom which will require you figuring that out most likely.

**Reply by hello-smile6** [reply]
Got it!

**Comment by james-pre** [comment]
Hi @hello-smile6, I'm the maintainer of BrowserFS. I'd like to let you know that BrowserFS has been deprecated. You can read more about that on the BrowserFS readme (https://github.com/jvilk/BrowserFS). [ZenFS](https://github.com/zen-fs/core) is an updated and active fork that I created.
